File: cpp/1827.minimum-operations-to-make-the-array-increasing.cpp
/*
 * @lc app=leetcode id=1827 lang=cpp
 *
 * [1827] Minimum Operations to Make the Array Increasing
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    int minOperations(vector<int> &nums)
    {
        int res = 0;
        int prev = 0;
        for (auto n : nums)
        {
            if (n > prev)
            {
                prev = n;
            }
            else
            {
                res += prev - n + 1;
                ++prev;
            }
        }
        return res;
    }
};
// @lc code=end

File: py/1827.minimum-operations-to-make-the-array-increasing.py
#
# @lc app=leetcode id=1827 lang=python3
#
# [1827] Minimum Operations to Make the Array Increasing
#

# @lc code=start
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        res = 0
        prev = 0
        for n in nums:
            if n > prev:
                prev = n
            else:
                res += prev-n+1
                prev += 1
        return res
# @lc code=end
